src/shared/watchdog.c

```diff
@@ -17,8 +17,36 @@
 static int watchdog_fd = -1;
 static char *watchdog_device = NULL;
 static usec_t watchdog_timeout = USEC_INFINITY;
+static usec_t watchdog_pretimeout = USEC_INFINITY;
 static usec_t watchdog_last_ping = USEC_INFINITY;
 
+static int update_pretimeout(void) {
+        int r;
+
+        if (watchdog_fd < 0)
+                return 0;
+
+        if (watchdog_pretimeout == USEC_INFINITY || watchdog_timeout == USEC_INFINITY)
+                return 0;
+        else {
+                char buf[FORMAT_TIMESPAN_MAX];
+                int sec = (int) DIV_ROUND_UP(watchdog_pretimeout, USEC_PER_SEC);
+                r = ioctl(watchdog_fd, WDIOC_SETPRETIMEOUT, &sec);
+                if (r < 0) {
+                        /* EOPNOTSUPP means the watchdog does not support pretimeouts */
+                        log_full(ERRNO_IS_NOT_SUPPORTED(errno) ? LOG_DEBUG : LOG_WARNING,
```

Comment:
Should this be a debug or an info message? Now that I think about it, you might want to know if you were trying to set a pretimeout but your watchdog didn't support it.

Reply:
given it should only happen once on boot (or reexec), it should be fine to make it 'info'